src/sync/SyncModal.js

```diff
@@ -16,33 +16,34 @@ import {
     PROGRESS_LOADING,
 } from './constants';
 
-const getStatusMessage = (progress, total, isSyncing, errorMessage, progressMessage) => {
-  let message = '';
+export function SyncModal({ database, isOpen, onClose, onPressManualSync, state }) {
+  const getStatusMessage = (progress, total, isSyncing, errorMessage, progressMessage) => {
+    let message = '';
 
-  if (errorMessage !== '') {
-    message = errorMessage;
-  } else if (!isSyncing) {
-    message = 'Sync Complete.';
-  } else if (progress >= total) {
-    message = 'All records updated.';
-  } else if (progress === PROGRESS_LOADING) {
-    message = 'Loading change count...';
-  } else {
-    message = progressMessage ? `${progressMessage}\n` : '';
-    message += `${progress} of ${formatPlural('@count record', '@count records', total)} updated`;
-  }
+    if (errorMessage !== '') {
+      message = errorMessage;
+    } else if (!isSyncing) {
+      const recordsToSyncCount = database.objects('SyncOut').length;
+      message = recordsToSyncCount > 0 ? `${recordsToSyncCount} Records Waiting` : 'Sync Complete';
```

Comment:
This is main change, most of this diff is from moving the function inside the SyncModal function to get scope of the database prop (added in this commit)

Reply:
@edmofro might like to do something similar in meditrak. So you are tagged.